mteb/models/cache_wrapper.py

```diff
@@ -0,0 +1,330 @@
+                for hash_, index in self.hash_to_index.items()
+            }
+
+            with open(self.index_file, "w", encoding="utf-8") as f:
+                json.dump(serializable_index, f, indent=2)
+
+            self._save_dimension()
+            logger.info(f"Saved TextVectorMap to {self.directory}")
+        except Exception as e:
+            logger.error(f"Error saving TextVectorMap: {str(e)}")
+            raise
+
+    def load(self, name: str = None) -> None:
+        name_details = name if name else ""
+        try:
+            self._load_dimension()
+            if self.index_file.exists() and self.vectors_file.exists():
+                with open(self.index_file, encoding="utf-8") as f:
+                    # Load and convert the JSON data back to the expected format
+                    loaded_index = json.load(f)
+                    self.hash_to_index = {
+                        str(hash_): int(index)  # Ensure we maintain the correct types
+                        for hash_, index in loaded_index.items()
+                    }
+
+                if self.vector_dim is not None:
+                    self.vectors = np.memmap(
+                        self.vectors_file, dtype="float32", mode="r+"
+                    )
+                    self.vectors = self.vectors.reshape(-1, self.vector_dim)
+                    logger.info(f"Loaded vectors file with shape: {self.vectors.shape}")
+                else:
+                    logger.warning(
+                        "Vector dimension not set. Unable to load vectors file."
+                    )
+
+                logger.info(
+                    f"Loaded TextVectorMap ({name_details}) from {self.directory}"
+                )
+            else:
+                logger.warning(
+                    f"No existing files found. Initialized empty TextVectorMap ({name_details})."
+                )
+        except Exception as e:
+            logger.error(f"Error loading TextVectorMap ({name_details}): {str(e)}")
+            raise
+
+    def get_vector(self, text: str) -> Optional[np.ndarray]:
+        try:
+            text_hash = self._hash_text(text)
+            if text_hash not in self.hash_to_index:
+                logger.debug(f"Text hash not found in index: {text_hash}")
+                return None
+            index = self.hash_to_index[text_hash]
+            return self.vectors[index]
+        except Exception as e:
+            logger.error(f"Error retrieving vector for text: {str(e)}")
+            raise
+
+    def __contains__(self, text: str) -> bool:
+        return self._hash_text(text) in self.hash_to_index
+
+    def __del__(self):
+        self.close()
+
+    def close(self):
+        if hasattr(self, "vectors") and self.vectors is not None:
+            self.vectors.flush()
+            del self.vectors
+            self.vectors = None
+        logger.info(f"Closed TextVectorMap in directory: {self.directory}")
+
+
+class CachedEmbeddingWrapper:
```

Comment:
```suggestion
class CachedEmbeddingWrapper(Encoder):
```

Reply:
I've also created a dummy class for wrappers to indicate that the model is already wrapped. I think any new wrapper classes should inherit from this as well.

Reply:
Will you link to it

Reply:
https://github.com/embeddings-benchmark/mteb/blob/main/mteb/models/wrapper.py

Reply:
Gotcha, so inherit from Wrapper, Encoder then!